ui/settings_page.go

```diff
@@ -2,36 +2,357 @@ package ui
 
 import (
 	"gioui.org/layout"
+	"gioui.org/widget"
 
 	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/values"
+	"github.com/planetdecred/godcr/wallet"
 )
 
-const PageSettings = "SettingsPage"
+const PageSettings = "Settings"
 
 type settingsPage struct {
-	dummyText decredmaterial.Label
+	pageContainer layout.List
+	theme         *decredmaterial.Theme
+	walletInfo    *wallet.MultiWalletInfo
+	wal           *wallet.Wallet
+
+	currencyConversion decredmaterial.IconButton
+	connectToPeer      decredmaterial.IconButton
+	userAgent          decredmaterial.IconButton
+	changeStartupPass  decredmaterial.IconButton
+
+	spendUnconfirm  *widget.Bool
+	startupPassword *widget.Bool
+	notificationW   *widget.Bool
+
+	line *decredmaterial.Line
+
+	isStartupPassword bool
+	errChann          chan error
 }
 
 func (win *Window) SettingsPage(common pageCommon) layout.Widget {
+	icon := common.icons.chevronRight
 	pg := &settingsPage{
-		dummyText: common.theme.H5("Not yet implemented"),
+		pageContainer: layout.List{
+			Axis: layout.Vertical,
+		},
+		theme:      common.theme,
+		walletInfo: win.walletInfo,
+		wal:        common.wallet,
+
+		spendUnconfirm:  new(widget.Bool),
+		startupPassword: new(widget.Bool),
+		notificationW:   new(widget.Bool),
+
+		line:     common.theme.Line(),
+		errChann: common.errorChannels[PageSettings],
+
+		currencyConversion: common.theme.PlainIconButton(new(widget.Clickable), icon),
+		connectToPeer:      common.theme.PlainIconButton(new(widget.Clickable), icon),
+		userAgent:          common.theme.PlainIconButton(new(widget.Clickable), icon),
+		changeStartupPass:  common.theme.PlainIconButton(new(widget.Clickable), icon),
 	}
+	pg.line.Height = 2
+	pg.line.Color = common.theme.Color.Background
+
+	color := common.theme.Color.LightGray
+	zeroInset := layout.UniformInset(values.MarginPadding0)
+
+	pg.currencyConversion.Color, pg.currencyConversion.Inset = color, zeroInset
+	pg.connectToPeer.Color, pg.connectToPeer.Inset = color, zeroInset
+	pg.userAgent.Color, pg.userAgent.Inset = color, zeroInset
+	pg.changeStartupPass.Color, pg.changeStartupPass.Inset = color, zeroInset
 
 	return func(gtx C) D {
 		pg.handle(common)
 		return pg.Layout(gtx, common)
 	}
 }
 
-// main settings layout
 func (pg *settingsPage) Layout(gtx layout.Context, common pageCommon) layout.Dimensions {
-	return common.Layout(gtx, func(gtx C) D {
-		return layout.Center.Layout(gtx, func(gtx C) D {
-			return pg.dummyText.Layout(gtx)
+	pg.updateSetting()
+	body := func(gtx C) D {
+		page := SubPage{
+			title:      "Settings",
+			walletName: "",
```

Comment:
You don't need to pass an empty string to the field. You can simply leave the field out